tools/api.py

```diff
@@ -0,0 +1,414 @@
+"""Produce a list of API endpoints that are dialed by AoU endpoints.
+
+This program requires two command-line parameters:
+
+    1. The path to the root of an AoU code repository
+    2. The path to a file containing a call graph of the AoU Java API code
+
+The call graph file is formatted as described here: https://github.com/gousiosg/java-callgraph
+
+To generate the call graph, do the following:
+
+```
+git clone https://github.com/gousiosg/java-callgraph.git
+cd path/to/java-callgraph
+mvn install # install 'maven' before running this command
+java -jar target/javacg-0.1-SNAPSHOT-static.jar path-to-aou-repo/{aou-utils/swagger-codegen-cli.jar,common-api/build/libs/common-api-0.1.0.jar} >path/to/callgraph.txt
+```
+
+Then, to produce the list of endpoints, run `python3 api.py path/to/aou-repo path/to/callgraph.txt`. The endpoints are output in CSV.
```

Comment:
Nice documentation on how to use this. Love it.

Reply:
Thanks!